lib/tokens/IamTokenManager.js

```diff
@@ -90,6 +90,13 @@ class IAMTokenManager extends TokenManager {
     ], (error, responses) => {
       done(error, responses[responses.length - 1]);
     });
+    setTimeout(function() {
+      debug('Destroying IAM token requests');
+      reqToken.abort();
```

Comment:
Can you add a comment here please explaining why this is necessary (referencing the memory leak etc)?

Reply:
Instead of this approach I created a new Agent that has no `keepAlive` set to `true` This way every token refreshment request will be removed from the memory when they completed.